### ISSUE TITLE
Install a later version of AWS CLI v1

### ISSUE THREAD

**Issue by paulca** [buildkite/elastic-ci-stack-for-aws#1027]
**Is your feature request related to a problem? Please describe.**
This is no longer a problem for me because I've since modified my pipeline to run the AWS CLI inside a Docker container, which is probably a much better way to do this anyway.

However, in case it helps anyone, there have been a number of updates to the AWS CLI, one in particular that broke our deploys because it was incompatible with an update to EKS. A later version of 1 would have worked. pip on Elastic Stack seems to install awscli 1.19, when the latest version on pypy is 1.24. I don’t know enough to know why it’s not installing the latest version.

**Describe the solution you'd like**
1.24 would have been totally fine.

**Describe alternatives you've considered**
Moving our deploy script inside a Docker container where we control the `awscli` version was a perfect solution for us, and in fact a much better solution overall because it meant that the `awscli` was project specific, rather than stack specific.

**Additional context**
It might be worth adding a note to the documentation recommending that folks actually run deploy scripts inside Docker containers if that's the preferred way of doing this.

**Comment by moskyb** [comment]
hey there @paulca! agreed, we totally need to be installing a later version of the AWS cli. Somewhat related, i think that when we do the next major version bump of the elastic stack, we'll probably only install the AWS CLI v2.

In re your specific issue, i thiiink the issue might be that we're [using pip2 (😬) to install the AWS cli](https://github.com/buildkite/elastic-ci-stack-for-aws/blob/master/packer/linux/scripts/install-utils.sh#L10). I suspect that if we just switch to using pip3 in this case (and ideally scrub python 2 from the elastic stack entirely), we should get the latest v1 (1.25 at time of writing).

**Comment by pda** [comment]
Somewhat relevant to staying on v1 until v6: https://github.com/buildkite/elastic-ci-stack-for-aws/pull/670#issuecomment-706702069

**Comment by admackin** [comment]
We would love to this upgrade too.

**Comment by admackin** [comment]
OK my workaround is to install a new version inside a virtualenv (avoiding extra work of using Docker)

If I source the following script I get the latest version when I run `aws` command

```
#!/bin/sh

# source this script using `. _updated_aws_cli` if you need a newer version of `aws` than Buildkite provides (1.19)

python3 -m venv .aws-cli-venv
. .aws-cli-venv/bin/activate
pip3 install awscli

function aws {
  # for some reason the new installed aws-cli version ignores #! line on Buildkite so we
  # work around like so
  python3 $(which aws) $*
}
```

**Comment by CaseyLabs** [comment]
Following up on this issue: 

- Currently the Builkdite Elastic CI Amazon Linux ec2 image is running AWS CLI v1

- Upgrading the AWS CLI to v2 was held back until Elastic CI v5 was released [[original comment](https://github.com/buildkite/elastic-ci-stack-for-aws/pull/670#issuecomment-706702069)]

Any chance we can get the AWS CLI upgraded to v2?

cc @pda @yob

**Comment by SoManyHs** [comment]
Should this be closed? I saw it was fixed in #1131.

**Comment by triarius** [comment]
Thanks for the catch @SoManyHs
